great_expectations/datasource/fluent/directory_data_asset.py

```diff
@@ -0,0 +1,68 @@
+from __future__ import annotations
+
+import logging
+import pathlib
+from typing import TYPE_CHECKING
+
+from great_expectations.core.batch import BatchDefinition
+from great_expectations.core.id_dict import IDDict
+from great_expectations.datasource.fluent.constants import (
+    _DATA_CONNECTOR_NAME,
+)
+from great_expectations.datasource.fluent.file_path_data_asset import _FilePathDataAsset
+
+if TYPE_CHECKING:
+    from great_expectations.datasource.fluent.batch_request import (
+        BatchRequest,
+    )
+
+logger = logging.getLogger(__name__)
+
+
+class _DirectoryDataAsset(_FilePathDataAsset):
```

Comment:
This intermediate class is used to standardize the directory specific param `data_directory` and the directory specific `_get_batch_definition_list()` logic.

Reply:
@anthonyburdi Should `_DirectoryDataAsset` really inherit  `_FilePathDataAsset`?  Or should they be siblings?  I worry that there are properties that will be inherited, which do not belong.  Could you please weigh in?  Thank you.